File: src/main/java/es/santander/ascender/retoGrupoCIC/model/Item.java
package es.santander.ascender.retoGrupoCIC.model;

import java.time.LocalDate;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

@Entity
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nombre;

    @ManyToOne
    private TipoItem tipo;

    @ManyToOne
    private Formato formato;

    private String ubicacion;
    
    private LocalDate fecha;

    @Enumerated(EnumType.STRING)
    private EstadoItem estado = EstadoItem.DISPONIBLE;
 
}

File: front/src/item.ts
export interface Item {
  id: number;
  nombre: string;
  ubicacion: string;
  tipo: {
    id: number;
    nombre: string;
  };
  formato: {
    id: number;
    nombre: string;
  };
  fecha: string;
  estado: string;
  }
  